geolocation.py

```diff
@@ -1,10 +1,23 @@
 
 class Geolocation:
 
-    def __init__(self,country,region,city,latitude,longitude,adSetId = None):
+    def __init__(self,country=None,region=None,city=None,latitude=None,longitude=None,adSetId=None,engine=None,metadata=None):
```

Comment:
You can probably break this line (as it's a bit long):
```suggestion
    def __init__(self, country=None, region=None, city=None, latitude=None, longitude=None, adSetId=None, engine=None, metadata=None):
```